### PR TITLE
syncer: remove unreasonable context.Background()

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/1845

### What is changed and how it works?
replace `Query` with `QueryContext`

### Check List 

Tests

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -2894,7 +2895,6 @@ func (s *Syncer) printStatus(ctx context.Context) {
 				}
 			}
 
-			ctx2, cancel2 := context.WithTimeout(ctx, utils.DefaultDBTimeout)
 			latestMasterPos, latestmasterGTIDSet, err = s.getMasterStatus(ctx2)
```
Comment (GMHDBJD) [inline syncer/syncer.go:2898]: Maybe use another context ctx3?

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: syncer/syncer.go (hunk 1 above)
Comment (lichunzhu) [reply syncer/syncer.go:2898]:
```suggestion
			ctx2, cancel2 = context.WithTimeout(ctx, utils.DefaultDBTimeout)
```

Comment (lance6716) [reply syncer/syncer.go:2898]: call previous cancel2 before it's overwritten

Comment (okJiang) [reply syncer/syncer.go]: fixed, PTAL

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**